nav2_regulated_pure_pursuit_controller/include/nav2_regulated_pure_pursuit_controller/regulated_pure_pursuit_controller.hpp

```diff
@@ -260,12 +273,17 @@ class RegulatedPurePursuitController : public nav2_core::Controller
   double rotate_to_heading_min_angle_;
   double goal_dist_tol_;
   bool allow_reversing_;
+  double footprint_cost_;
+  double possible_inscribed_cost_{-1};
```

Comment:
Remove this and all the logic around it -- that's specific to planning because we're checking upwards of 1M nodes, here its like 20, so can just check use the actual footprint for all of them.